src/components/Breadcrumb.vue

```diff
@@ -0,0 +1,107 @@
+<script>
+export default {
+  props: {
+    location: {
+      type: Object,
+      required: true,
+      validator: val => val.type && val.id,
+    },
+    loading: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    append: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  inject: ['girderRest'],
+  asyncComputed: {
+    breadcrumb: {
+      default: { root: {}, path: [] },
+      async get() {
+        this.$emit('update:loading', true);
+        const breadcrumb = { root: {}, path: [] };
+        const { id, type, name } = this.location;
+        if (type === 'folder') {
+          // The last breadcrumb isn't returned by rootpath.
+          if (name) {
+            breadcrumb.path.unshift(this.location);
+          } else {
+            const { data } = await this.girderRest.get(`folder/${id}`);
+            breadcrumb.path.unshift(this.extractCrumbData(data));
+          }
+          // Get the rest of the path.
+          const { data } = await this.girderRest.get(`folder/${id}/rootpath`);
+          data.reverse().forEach((crumb) => {
+            const { object } = crumb;
+            const entity = this.extractCrumbData(object);
+            if (entity.type === 'folder') {
+              breadcrumb.path.unshift(entity);
+            } else {
+              breadcrumb.root = entity;
+            }
+          });
+        } else {
+          const { data } = await this.girderRest.get(`${type}/${id}`);
+          breadcrumb.root = this.extractCrumbData(data);
+        }
+        this.$emit('update:loading', false);
+        return breadcrumb;
+      },
+    },
+  },
+  methods: {
+    extractCrumbData(object) {
+      return {
+        name: object._modelType !== 'user' ? object.name : object.login,
+        id: object._id,
+        type: object._modelType,
+        parentId: object.parentId,
+        parentType: object.parentCollection,
+      };
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+v-breadcrumbs.girder-breadcrumb-component.pl-0.pt-0.pb-0
+  v-icon.mdi-24px(
+      :class="{ disabled: disabled }",
+      slot="divider",
+      color="accent") {{ $vuetify.icons.chevron }}
+  v-breadcrumbs-item(
+      :class="{ disabled: disabled }",
+      @click.native="$emit('changelocation', breadcrumb.root)")
+    v-icon.mdi-24px(color="accent") {{ $vuetify.icons.globe }}
+    | &nbsp; {{ breadcrumb.root.name }}
+  v-breadcrumbs-item(
+      :class="{ disabled: disabled }",
+      v-for="item in breadcrumb.path",
+      :key="`${item.id}.crumb`",
+      @click.native="$emit('changelocation', item)") {{ item.name }}
```

Comment:
`@click` doesn't work on any Vuetify components because they aren't native DOM elements.

I think elements like v-btn step around this by forwarding native events to the parent -- see https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/components/VBtn/VBtn.ts#L104

For whatever reason, the event isn't forwarded for v-breadcrumb-item, so we can bind to the root element's events directly with `.native`.

https://github.com/vuetifyjs/vuetify/issues/4730